src/Documentation/DocumentationGenerator/ProcessDocComments.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Quantum.QsCompiler;
+using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.Documentation;
+using Microsoft.Quantum.QsCompiler.SyntaxTokens;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.Core;
+using Newtonsoft.Json.Linq;
+
+using Range = Microsoft.Quantum.QsCompiler.DataTypes.Range;
+
+#nullable enable
+
+namespace Microsoft.Quantum.Documentation
+{
+    /// <summary>
+    ///     A syntax tree transformation that parses documentation comments,
+    ///     saving documentation content back to the syntax tree as attributes,
+    ///     and writing formatted documentation content out to Markdown files.
+    /// </summary>
+    public class ProcessDocComments
+    : SyntaxTreeTransformation<ProcessDocComments.TransformationState>
+    {
+        public class TransformationState
+        { }
+
+        private readonly DocumentationWriter? writer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProcessDocComments"/> class.
+        /// </summary>
+        /// <param name="outputPath">
+        ///     The path to which documentation files should be written.
+        /// </param>
+        /// <param name="packageName">
+        ///     The name of the NuGet package being documented, or <c>null</c>
+        ///     if the documentation to be written by this object does not
+        ///     relate to a particular package.
+        /// </param>
+        public ProcessDocComments(
+            string? outputPath = null,
+            string? packageName = null
+        )
+        : base(new TransformationState())
+        {
+            this.writer = outputPath == null
+                          ? null
+                          : new DocumentationWriter(outputPath, packageName);
+
+            // We provide our own custom namespace transformation, and expression kind transformation.
+            this.Namespaces = new ProcessDocComments.NamespaceTransformation(this, this.writer);
```

Comment:
Could you please disable all other transformations? I think since you need the namespace transformation only, you can likely also just disable everything in the base ("base(new TransformationState(), TransformationOptions.Disabled)").

Reply:
Will do, let me apply that fix. Thanks!